From 5694ab26d67c8360975f5b1fbec0418dad0c5348 Mon Sep 17 00:00:00 2001
From: Chao Sun <sunchao@apache.org>
Date: Tue, 28 Feb 2017 15:18:56 -0800
Subject: [PATCH 1089/1164] CDH-53209 HIVE-16060: GenericUDTFJSONTuple's json
 cache could overgrow beyond its limit (Chao Sun,
 reviewed by Xuefu Zhang)

Change-Id: Ib4f94ed82131ff2380b3bfc0460f54a9da9d57a0
---
 .../hive/ql/udf/generic/GenericUDTFJSONTuple.java  |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFJSONTuple.java b/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFJSONTuple.java
index ea5aeec..dda476f 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFJSONTuple.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFJSONTuple.java
@@ -91,7 +91,7 @@ protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
 
   }
 
-  static Map<String, Object> jsonObjectCache = new HashCache<String, Object>();
+  private transient Map<String, Object> jsonObjectCache;
 
   @Override
   public void close() throws HiveException {
@@ -103,6 +103,7 @@ public StructObjectInspector initialize(ObjectInspector[] args)
 
     inputOIs = args;
     numCols = args.length - 1;
+    jsonObjectCache = new HashCache<>();
 
     if (numCols < 1) {
       throw new UDFArgumentException("json_tuple() takes at least two arguments: " +
-- 
1.7.9.5

